### PR TITLE
SPIDER: Chicago Park District Board of Commissioners

### PR DESCRIPTION
**Pull request by bergren2**
Basically a rewrite of #42 to use the Legistar scraper, plus correct branch names and whatnot.

Bonus: Added [Freezegun](http://stevepulec.com/freezegun/) for writing tests around time

### CODE REVIEW

**Comment by bergren2** [comment]
Okeydoke, I think this is ready for the review gauntlet!

File: documenters_aggregator/spiders/parkdistrict.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+        """
+        return 'Not classified'
+
+    def _parse_status(self, item, start_time):
+        """
+        passed = meeting already started
+        tentative = no agenda posted
+        confirmed = agenda posted
+        """
+        if datetime.now().isoformat() > start_time:
```
Comment (jim) [inline documenters_aggregator/spiders/parkdistrict.py]: I am a little worried about using `status` to indicate two different things: whether a meeting has already started and if an agenda has been posted. Using the presence of an agenda to determine if a meeting is confirmed is clever but not totally obvious to those consuming the data.

I am OK with merging this as is, though, while we think about how to handle this better. I think we probably ought to have a place for an agenda URL in the event schema. Additionally, whether the meeting has started of now can be inferred from the `start_time`, so altering this value based the current time doesn't provide a lot of value and creates churn in the data.

I have created #169 to track making those schema changes, at which point we can come back and update this spider.

File: tests/utils.py [hunk 2]
```diff
@@ -15,6 +16,7 @@ def file_response(file_name, url=None):
     if not url:
         url = 'http://www.example.com'
 
+    urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
```
[the remark this answers is not in the document]
Comment (bergren2) [reply tests/utils.py]: No warnings before adding my spider, which was weird. I can dig into this a little bit more, but the error I was seeing when running tests was this:

```
collecting 6244 items/home/travis/virtualenv/python3.6.3/lib/python3.6/site-packages/urllib3/connectionpool.py:858: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
```

**Comment by bergren2** [comment]
@jim That's removing one of the `.DS_Store` files since I noticed it b/c it changed, looks like there's one more in the root. Want me to remove that one too and add the `.gitignore` change as part of this PR?

File: tests/utils.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (bergren2) [reply tests/utils.py]: Here's more context, from one of the previous builds: https://travis-ci.org/City-Bureau/documenters-aggregator/jobs/294139002

**Comment by bergren2** [comment]
@jim I moved the warning message suppression to my spider -- it was specifically the calls I was making with the [Legistar helper](https://github.com/opencivicdata/python-legistar-scraper). Also looks like there's been some more recent activity on that project so I can take a look at that tomorrow, along with updating my master to take care of the conflicting `.gitignore` file.

**Comment by bergren2** [comment]
Tracked down the original Legistar scraper decision to not verify requests to [this PR](/opencivicdata/scrapers-us-municipal#81):

> Under the hood requests (which uses urllib3 for SSL junk I'm pretty sure) is not allowing us to load pages from the Chicago Legistar site because it is forcing SSL while not really doing the best job at supporting a secure connection.

There are a bunch of suggestions for disabling the error messages in [this thread on the requests library](/requests/requests#2214) -- my main concern with suppressing the warnings in the first place was to not add superfluous messaging to the logs, so I'm okay with un-suppressing them if we want to go that route.

**Comment by jim** [comment]
@bergren2 Ah, so it's because Legistar's SSL setup has issues. I think it makes sense to then disable the warning on a per-spider basis for now. For our uses, there isn't a huge risk coming from not using/validating SSL.

I mis-read the `.DS_Store` removal above as an addition! Thanks for removing it, and I'll take care of any other ones that snuck into the repo.

**Comment by diaholliday** [comment]
Hey @jim @r-wei can either of you update me on the current status on this spider?

**Comment by r-wei** [comment]
@diaholliday it's finished and running with the rest. I see there are 2 events for the Chicago Park District in the EventsTest table.

**Comment by diaholliday** [comment]
Ahh, whoops, you're right. I missed it in my first search

**Pull request merged**